core/src/main/java/hudson/model/PeriodicWork.java

```diff
@@ -99,15 +102,49 @@ public long getInitialDelay() {
     @Initializer(after= JOB_LOADED)
     public static void init() {
         // start all PeriodicWorks
-        for (PeriodicWork p : PeriodicWork.all()) {
-            Timer.get().scheduleAtFixedRate(p, p.getInitialDelay(), p.getRecurrencePeriod(), TimeUnit.MILLISECONDS);
+        ExtensionList<PeriodicWork> extensionList = all();
+        extensionList.addListener(new PeriodicWorkExtensionListListener(extensionList));
+        for (PeriodicWork p : extensionList) {
+            schedulePeriodicWork(p);
         }
     }
 
+    private static void schedulePeriodicWork(PeriodicWork p) {
```

Comment:
as above

Reply:
private should not need javadoc or since?

Reply:
Right, missed the different modifier.